diff -Nurd --exclude-from=diff_ignore_files.txt nethack-3.4.3-orig/include/flag.h nethack-3.4.3-showbuc/include/flag.h
--- nethack-3.4.3-orig/include/flag.h	2003-12-08 01:39:13.000000000 +0200
+++ nethack-3.4.3-showbuc/include/flag.h	2006-05-15 22:35:10.000000000 +0300
@@ -264,6 +264,7 @@
 	boolean wc2_softkeyboard;	/* use software keyboard */
 	boolean wc2_wraptext;		/* wrap text */
 
+	boolean show_buc;	/* always show BUC status */
 	boolean  cmdassist;	/* provide detailed assistance for some commands */
 	boolean	 obsolete;	/* obsolete options can point at this, it isn't used */
 	/* Items which belong in flags, but are here to allow save compatibility */
diff -Nurd --exclude-from=diff_ignore_files.txt nethack-3.4.3-orig/src/objnam.c nethack-3.4.3-showbuc/src/objnam.c
--- nethack-3.4.3-orig/src/objnam.c	2003-12-08 01:39:13.000000000 +0200
+++ nethack-3.4.3-showbuc/src/objnam.c	2006-05-15 22:40:00.000000000 +0300
@@ -600,7 +600,7 @@
 		Strcat(prefix, "cursed ");
 	    else if (obj->blessed)
 		Strcat(prefix, "blessed ");
-	    else if ((!obj->known || !objects[obj->otyp].oc_charged ||
+	    else if (iflags.show_buc || (!obj->known || !objects[obj->otyp].oc_charged ||
 		      (obj->oclass == ARMOR_CLASS ||
 		       obj->oclass == RING_CLASS))
 		/* For most items with charges or +/-, if you know how many
diff -Nurd --exclude-from=diff_ignore_files.txt nethack-3.4.3-orig/src/options.c nethack-3.4.3-showbuc/src/options.c
--- nethack-3.4.3-orig/src/options.c	2003-12-08 01:39:13.000000000 +0200
+++ nethack-3.4.3-showbuc/src/options.c	2006-05-15 22:40:07.000000000 +0300
@@ -165,6 +165,7 @@
 #else
 	{"showexp", (boolean *)0, FALSE, SET_IN_FILE},
 #endif
+	{"showbuc", &iflags.show_buc, FALSE, SET_IN_GAME},
 	{"showrace", &iflags.showrace, FALSE, SET_IN_GAME},
 #ifdef SCORE_ON_BOTL
 	{"showscore", &flags.showscore, FALSE, SET_IN_GAME},
